main/src/addins/MonoDevelop.TextEditor/MonoDevelop.TextEditor/TextViewDisplayBinding.cs

```diff
@@ -68,20 +69,25 @@ protected override IEnumerable<DocumentControllerDescription> GetSupportedContro
 			//".html",
 			//".js",
 			//".json",
-			//".ts",
-			//".xaml"
+			//".ts"
 		};
 
 		bool IsSupportedFileExtension (FilePath fileName)
 		{
 			return supportedFileExtensions.Contains (fileName.Extension);
 		}
 
-		bool IsSupportedAndroidFileName (FilePath fileName, WorkspaceObject ownerProject)
+		bool IsSupportedDesignerFileName (FilePath fileName, WorkspaceObject ownerProject)
 		{
-			// disable Android XML for now
-			return false;
+			if (!FeatureSwitchService.IsFeatureEnabled ("DesignersNewEditor").GetValueOrDefault ())
+				return false;
+
+			return string.Equals (".xaml", fileName.Extension, StringComparison.OrdinalIgnoreCase)
```

Comment:
Maybe
```suggestion```
			return fileName.HasExtension (".xaml")